### PR TITLE
catch attempts to write outside the media folder

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
## Purpose / Description
Main purpose: having libanki similar to Anki's folder Anki.

## Fixes
Theoretical risk in case of media name in cards not referring to media folder

## Approach
As in Anki

## How Has This Been Tested?

I tried to import a deck with an image. it did work.
I didn't check whether it really protects against a deck created especially to hack anki/ankidroid, since I don't even know exactly against which attack this is supposed to protect the user. I don't know how to create a zip file such that the normalized path of a string is above the top level of the folder.

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

**Comment by Arthur-Milchior** [comment]
A single fail; I fear it must be restarted. Can you please do it ?

**Comment by Arthur-Milchior** [comment]
Tabs corrected.
Codacy does not like a runtimeexception, I fear I've got nothing best to offer.
Top message explaining what testing is done

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/AnkiPackageImporter.java [hunk 1]
```diff
@@ -89,16 +89,22 @@ public void run() throws ImportExportException {
...
                 jr.beginObject();
-                String name;
-                String num;
+                String name; // v in anki
+                String num; // k in anki
                 while (jr.hasNext()) {
                     num = jr.nextName();
                     name = jr.nextString();
+                    String path = new File(dir, name).getAbsolutePath();
+                    if (!path.startsWith(dir)) {
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/libanki/importer/AnkiPackageImporter.java]: We actually have a utility for this from a [previous PR](https://github.com/ankidroid/Anki-Android/pull/4860) - could reshape to use it so we know the logic is identical? (for instance, we use getCanonicalPath there vs getAbsolutePath here, and we should use the same - using the same method will enforce that)

https://github.com/ankidroid/Anki-Android/blob/master/AnkiDroid/src/main/java/com/ichi2/libanki/Utils.java#L631

I haven't thought how to handle the different exception type the utility throws (IOException) but having it be handled - even if just logged and ignored - vs Runtime and thus blowing up the app seems cleaner?

**Comment by Arthur-Milchior** [comment]
It's now pushed using your method. I then tested again importing a package

**Comment by timrae** [comment]
There's an unused variable according to codacy. Also a typo in the exception if you feel like fixing it lol.

**Comment by Arthur-Milchior** [comment]
Done
My bad. The typo was not even here in Anki's commit I was porting

**Comment by mikehardy** [comment]
All the other exceptions in this method are handled though - with an error log and maybe user prompt? I didn't see anything else in the method that would throw RuntimeException and thus bubble up and I think crash the app. Why crash the app with this one instead of handling it like the others?

**Comment by Arthur-Milchior** [comment]
I simply did as Anki does in apkg.py l 35.  I tried to remain as close as possible to anki original.

Anyway, if this error occurs, it's probable that the purpose of the deck was to hack the computer/phone. That's not something that could occurs normally; but that's something that could be use to put data outside of the media folder. (See original Anki's commit to see more details).
So I believe that importing even part of such a deck would be a bad idea.

**Comment by Arthur-Milchior** [comment]
Thank you for letting me know.
If I receive a clear indication about what I should do wen this problem occur, I'll code it (if I can).
Until there is a decision made, I'll leave the code as it currently is.

**Comment by timrae** [comment]
It seems there is also one behaviour change here by adding the NFC normalisation. Can you please add that as a separate commit to make  debugging easier later if it causes any problems?

**Comment by Arthur-Milchior** [comment]
Done. Assuming I understood what you were asking

**Pull request merged**